python/dolfinx/wrappers/fem.cpp

```diff
@@ -148,6 +148,10 @@ void fem(py::module& m)
 
   m.def("create_sparsity_pattern", &dolfinx::fem::create_sparsity_pattern,
         "Create a sparsity pattern for bilinear form.");
+  m.def("pack_coefficients", &dolfinx::fem::pack_coefficients,
```

Comment:
This won't be very efficient because the C++ side returns at `std::vector`, which can't be wrapped as a NumPy array.

Reply:
Good point, `std::vector` is automatically casted to a `list` which is very slow in Python. This could cause a lot of hidden performance issues when passing `list` to a custom kernel.
You should probably make pack_foo return Eigen arrays - which is nicely returned as NumPy array in Python, or you can in pybind11 return a py::array_t with some hacks to avoid copying. I would prefer Eigen.

Reply:
@michalhabera I've proposed a solution, where we use eigen arrays to pack constants. Is there any more efficient way of inserting the std::vectors into the Eigen arrays that I've missed?

Reply:
@jorgensd The best way is to use Eigen::Map and map a continuous chunk of memory as Eigen::Array. You can find example in many places in dolfinx. (so just prepare std::vector as before, but in the very last line return an Eigen::Map)

Reply:
I do not think pac_constants is performance critical in preparation of the eigen array, but personally I find Eigen:Mapping nice and readable.

Reply:
latest commit uses Eigen::Map  as I agree it is very readable.